pkg/vsphere/infrastructure/ensurer/ensurer.go

```diff
@@ -82,14 +78,23 @@ func NewNSXTInfrastructureEnsurer(logger logr.Logger, nsxtConfig *vinfra.NSXTCon
 	}, nil
 }
 
-func (e *ensurer) NewStateWithVersion() (*api.NSXTInfraState, error) {
+func (e *ensurer) NewStateWithVersion(overwriteVersion *string) (*api.NSXTInfraState, error) {
+	if overwriteVersion != nil {
+		if !sets.NewString(api.SupportedEnsurerVersions...).Has(*overwriteVersion) {
+			return nil, fmt.Errorf("invalid overwrite version %s", *overwriteVersion)
+		}
+		return &api.NSXTInfraState{
+			Version: overwriteVersion,
+		}, nil
+	}
+
 	nsxtVersion, err := getNSXTVersion(e.connector)
 	if err != nil {
 		return nil, err
 	}
-	version := Version1_NSXT25
+	version := api.Ensurer_Version1_NSXT25
 	if strings.HasPrefix(*nsxtVersion, "3.") {
```

Comment:
You should check for 2.X to be set to 1 and everything above to version 2

Reply:
done

Reply:
done